src/pages/cart/Item.tsx

```diff
@@ -0,0 +1,94 @@
+import { useEffect, useState } from "react";
+import { formatPrice } from "../../utils/format";
+import { CartItem } from "../../utils/types";
+import { SlArrowUp, SlArrowDown } from "react-icons/sl";
+import { useQueryClient } from "@tanstack/react-query";
+import { useUpdateCart } from "./useUpdateCart";
+import { useDeleteItem } from "./useDeleteItem";
+
+type ItemProps = {
+  item: CartItem;
+};
+
+function Item({ item }: ItemProps) {
+  const [quantity, setQuantity] = useState(item.quantity);
+  const queryClient = useQueryClient();
+
+  const { error: updateItemError, mutate: updateItem } = useUpdateCart(
+    {
+      ...item,
+      quantity,
+    },
+    {
+      onSuccess: () => {
```

Comment:
If invalidating the "cart" query is something that always needs to happen after updating the cart, I'd consider moving this inside the `useUpdateCart` hook to avoid repetition for callers of the hook and avoid bugs that could arise if a caller forgets to invalidate.